models/repo.go

```diff
@@ -1553,9 +1553,18 @@ func GetUserMirrorRepositories(userID int64) ([]*Repository, error) {
 }
 
 // GetRecentUpdatedRepositories returns the list of repositories that are recently updated.
-func GetRecentUpdatedRepositories(page, pageSize int) (repos []*Repository, err error) {
-	return repos, x.Limit(pageSize, (page-1)*pageSize).
-		Where("is_private=?", false).Limit(pageSize).Desc("updated_unix").Find(&repos)
+func GetRecentUpdatedRepositories(page, pageSize int, userID int64) (repos []*Repository, err error) {
+	repos, _, err = SearchRepositoryByName(&SearchRepoOptions{
+		Keyword:  "%",
```

Comment:
currently empty string searches are handled by the "ranger" function which happens to be ```GetRecentUpdatedRepositories``` for the Explore page which is how it was always handled so empty string search will be enabled either way, but I could change the query and eliminate the like clause for empty string searches to eliminate potential performance problems.